qiskit_nature/properties/second_quantization/electronic/integrals/two_body_electronic_integrals.py

```diff
@@ -164,7 +164,7 @@ def to_spin(self) -> np.ndarray:
         if self._basis == ElectronicBasis.SO:
             return self._matrices  # type: ignore
 
-        so_matrix = np.zeros([2 * s for s in self._matrices[0].shape])
+        so_matrix = np.zeros([2 * s for s in self._matrices[0].shape], dtype="complex")
```

Comment:
I agree with @woodsp-ibm's original comment, that this is duplicating the memory requirements unnecessarily in most cases.
Thus, this should be a configurable setting.

We can either:
1. add a setting in `qiskit_nature.settings`
2. add an `__init__` argument to these integrals (since the `OneBody` case should probably deal with this, too) which would then also need to be propagated down to the `ElectronicEnergy`

I think, I would prefer the former option.

@pbark any thoughts on this?

Reply:
Thanks for the reply. Is it needed at all for this PR to work as intended though? If not, I would like to leave this change out and consider this scenario as part of #665 instead

Reply:
```suggestion
        so_matrix = np.zeros([2 * s for s in self._matrices[0].shape])
```
After another offline check with Panos, we decided to keep this change out of this PR and instead introduce this as a configurable option as part of the operator-overhaul (#665)